src/main/java/com/fasterxml/jackson/core/util/BufferRecyclerPool.java

```diff
@@ -38,17 +38,31 @@
 public interface BufferRecyclerPool extends Serializable
 {
     /**
-     * Method called to acquire {@link BufferRecycler}; possibly
-     * (but necessarily) a pooled recycler instance (depends on implementation
-     * and pool state).
+     * Method called to acquire a {@link BufferRecycler} from this pool
+     * AND make sure it is linked back to this
+     * {@link BufferRecyclerPool} as necessary for it to be
+     * released (see {@link #releaseBufferRecycler}) later on after
+     * usage ends.
+     * Actual acquisition is done by a call to {@link #acquireBufferRecycler()}.
+     *<p>
+     * Default implementation calls {@link #acquireBufferRecycler()} followed by
+     * a call to {@link BufferRecycler#withPool}.
      *
      * @return {@link BufferRecycler} for caller to use; caller expected
      *   to call {@link #releaseBufferRecycler} after it is done using recycler.
      */
-    BufferRecycler acquireBufferRecycler();
+    default BufferRecycler acquireAndLinkBufferRecycler() {
+        return acquireBufferRecycler().withPool(this);
+    }
+
+    /**
+     * Method for sub-classes to implement for actual acquire logic; called
+     * by {@link #acquireAndLinkBufferRecycler()}.
+     */
+    abstract BufferRecycler acquireBufferRecycler();
```

Comment:
Nitpick: this `abstract` modifier isn't necessary and probably a leftover of the former implementation.

Reply:
Ah, thanks!